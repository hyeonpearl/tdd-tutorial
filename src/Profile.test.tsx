import { render, screen } from '@testing-library/react';
import Profile from './Profile';

describe('<Profile />', () => {
  it('matches snapshot', () => {
    const utils = render(<Profile username='hyeonpearl' name='최현철' />);
    expect(utils.container).toMatchSnapshot();
  });

  it('shows the props correctly', () => {
    render(<Profile username='hyeonpearl' name='최현철' />);
    screen.getByText('hyeonpearl');
    screen.getByText('(최현철)');
    screen.getByText(/최/);
  });
});
