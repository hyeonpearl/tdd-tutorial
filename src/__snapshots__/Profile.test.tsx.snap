// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <b>
      hyeonpearl
    </b>
     
    <span>
      (
      최현철
      )
    </span>
  </div>
</div>
`;
